viam-orb-slam3.go

```diff
@@ -352,7 +348,9 @@ func New(ctx context.Context,
 	if err != nil {
 		return nil, err
 	}
-	cancelCtx, cancelFunc := context.WithCancel(ctx)
+
+	// In a module, ctx is short-lived which can cause the data process to exit early
```

Comment:
Could you please expand on this? Isn't there a risk that the data process will never terminate, leading to a goroutine leak & the module hanging when told to sigterm if we don't thread through the original context?

Reply:
We talked about this in person. 

This might in theory be a risk if the module manager (in rdk) is unable to properly shut down all of the modules, however it should be killing all of the module processes after calling Close (which should end this context and kill the data process). I personally think that the RDK should expect that any module can do crazy things with the system resources (like spawn processes) and that the RDK should be responsible of taking care of any resources that were not properly taken care of (like these possible rogue processes) 

If we wish to engineer around this, I see three ways forward:
- Add a heartbeat system 
- wait till modularizationv2
- somehow find a way to communicate a context across gRPC boundaries (this may be difficult)

Reply:
James confirmed we shouldn't use the `New()` context after `New()` terminates: https://viam.atlassian.net/browse/RSDK-2586. 

After investigating other places where `ctx` gets passed:
https://github.com/viamrobotics/viam-orb-slam3/pull/29/files#diff-c70dd5ed993c1b53cd893d5a38fa4665c771456384a369b2a67b7a55ddb9fffeR380
https://github.com/viamrobotics/viam-orb-slam3/pull/29/files#diff-c70dd5ed993c1b53cd893d5a38fa4665c771456384a369b2a67b7a55ddb9fffeR384

I've confirmed that both functions (which do start long lived processes) don't use the context for those long lived processes:
https://pkg.go.dev/go.viam.com/utils/pexec#ManagedProcess (note: this is not a one shot process so the context is not used)

and the grpc.DialContext https://pkg.go.dev/google.golang.org/grpc#DialContext also doesn't use it after the connection is established in the mode we use.

As a result, this code is correct & I'll change the code to use context.Background()

Reply:
Sgtm! Thanks for summarizing the research!